Update activity contractor costs section

mgwalker opened the pull request:
### This pull request changes...
- updates the activity contractor costs section
- fixes the dev seed to have correct data
- closes #1522

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

mgwalker commented on api/seeds/development/apds.js:
```diff
@@ -407,11 +407,44 @@ exports.seed = async knex => {
+              { contractor_resource_id: j, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: j, hours: 0, rate: 0, year: 2020 },
+              { contractor_resource_id: k, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: k, hours: 0, rate: 0, year: 2020 }
```
The hourly data is required. When an APD is created, the API inserts all this as initial data, but our seed was leaving it out. 😬

jeromeleecms commented:
@mgwalker Not sure if this is related or not, but when I'm working in the sample APD and click on the first private contractor: 

![image](https://user-images.githubusercontent.com/34279968/58034898-707aca80-7adc-11e9-905a-988a4762ea12.png)

The screen blanks out. Confirmed the behavior on IE and Firefox. Subsequent private contractors don't have this behavior and it opens like normal:

jeromeleecms commented:
@mgwalker Also, when the hourly resource "Yes" selection is made, the "Done" button appear on the same line as the dollar value:

![image](https://user-images.githubusercontent.com/34279968/58035165-0dd5fe80-7add-11e9-8ca2-529971bd3a3d.png)

mgwalker commented:
@jeromeleecms The first one is due to faulty data in the sample APD. 😬 I've updated your APD if you want to try it again.  I'll reset everyone else's as well.

jeromeleecms commented:
Changes above look good, but encountering this:

![image](https://user-images.githubusercontent.com/34279968/58047768-c449db80-7afd-11e9-9bef-55f1957c1343.png)

Per Retro, posting here to determine if this is a bug or not before approval.

mgwalker commented:
@jeromeleecms Is the save error still happening for you? I can't get it to happen, but if it's happening consistently for you, maybe we can pair up sometime to look into it?

jeromeleecms commented:
@mgwalker still getting the error:

![image](https://user-images.githubusercontent.com/34279968/58134348-67bcee00-7bdb-11e9-816e-273aa1688737.png)

I'll ping you on Slack tomorrow to check your available to pair on this.

mgwalker commented on web/src/containers/activity/ContractorResourcesForm.js:
```diff
@@ -0,0 +1,244 @@
+    contractor.start
+  ]);
+
+  const endDate = useMemo(() => dateSplitter(contractor.end), [contractor.end]);
```
That is a sensible request, to which the answer is...  can I please not?  😆

`prettier` is doing this (based on line length, I think).  I can make the change manually in a different editor (I've got VS Code configured to automatically run prettier on every save), but it'll just revert the next time I poke round in this file. 😛 I've taken the attitude of not fighting with prettier anymore and just accept that it's weird at the edges.

mgwalker commented on web/src/containers/activity/ContractorResourcesForm.js:
```diff
@@ -0,0 +1,244 @@
+      </fieldset>
+
+      {contractor.hourly.useHourly ? (
+        <p className="ds-u-margin-bottom--0">
```
I *think* this was fixed in 9efce0f where I wrapped the hourly boxes in the `<p>` tags. Prior to that, they were naked and inline. 😬

mgwalker commented:
@jeromeleecms I can reproduce the save error. Looking into it!  I think it's related to the start/end dates...

mgwalker commented:
@jeromeleecms Looks like it's fixed!  Thanks for finding that.  😄

austinhernandez commented:
<img width="807" alt="Screen Shot 2019-05-28 at 12 55 53 PM" src="https://user-images.githubusercontent.com/23194720/58497736-9a6e6580-814a-11e9-9595-3235109c19fa.png">

 - [ ] use the spacing from key personnel for the summary section
 - [ ] system notification to let me know i'm about to delete an important thing. (It's kinda close to the edit button so I could see this bein' an accidental deletion.)

*Missing data design*
<img width="687" alt="Screen Shot 2019-05-28 at 12 55 37 PM" src="https://user-images.githubusercontent.com/23194720/58498399-f84f7d00-814b-11e9-8fb0-47ecf8716a1d.png">
I'd love to get @jameshupp 's thoughts on what the content should be for the empty fields in this section. Also, If it's a `null` response for date, we should get a `null` for all the missing and important fields.

 - [ ] final content for missing data in the input fields @jameshupp and @NAretakis @jeromeleecms 
 - [ ] add an `<emphasis>` style to the `dates not completed` / no value text
 - [ ] @jeromeleecms and @NAretakis is important to specify no value was put in these $ fields or is it ok to default to $0?

*Here is a screenshot of the labels from Key Personnel for cross-reference:*
<img width="732" alt="Screen Shot 2019-05-28 at 12 57 49 PM" src="https://user-images.githubusercontent.com/23194720/58497326-bb828680-8149-11e9-9562-31fd48b18975.png">
 - [ ] bold labels and add colons at the end of the label to match the contractor costs section
 - [ ] system notification to let me know i'm about to delete the section

austinhernandez commented:
I was wrong about spacing. Please disregard the above about spacing and look at this mockup!
small top padding, larger bottom padding between the summary block (no label) and the labeled fields. 

![image](https://user-images.githubusercontent.com/23194720/58503642-08208e80-8157-11e9-9704-82facc933b33.png)

mgwalker commented:
- fixed the vertical spacing between the contractor name and description of services
- added a confirmation prompt before deleting the contractor
- added placeholder text for when the name and description are empty
- created #1558 to capture changes needed for the key personnel section

jameshupp commented:
> I'd love to get @jameshupp 's thoughts on what the content should be for the empty fields in this section. Also, If it's a `null` response for date, we should get a `null` for all the missing and important fields.
> 
I think I need more info here. Are we treating these as errors, or are we just telling someone they're empty? Probably want to write them slightly differently depending on which.
